### PR TITLE
Task 50

### PR DESCRIPTION
**Pull request by artsiomMandrukevich**
# Description
Task50. Created 3 tests for validating `PUT`/`PATCH` /users endpoints.
Task60. Created 3 tests for validating `DELETE` /users endpoint.

## How to test / repro
Run `UsersPutPatchTest` class.

### Steps for reproducing:
1. Up the web-service in the docker container.
`docker run -d -p 7072:8082 coherentsolutions/rest-training:1.0`
2. Call `PUT/PATCH /users` endpoint with a user to update and new values and incorrect zipCode.
### Actual Result:
User is removed from the application.

### Expect Result:
And User is not updated.

### Steps for reproducing:
1. Up the web-service in the docker container.
`docker run -d -p 7072:8082 coherentsolutions/rest-training:1.0`
2. Call `PUT/PATCH /users` endpoint with a user to update and new values and required fields are missed (age, zipCode).
### Actual Result:
The response code is 400.
User is removed from the application.

### Expect Result:
The response code is 409.
And User is not updated.

### Steps for reproducing:
1. Up the web-service in the docker container.
`docker run -d -p 7072:8082 coherentsolutions/rest-training:1.0`
2. Call `DELETE /users` endpoint with only required fields in the payload (name, sex).
### Actual Result:
User is **not** deleted from the application.

### Expect Result:
User is deleted from the application.

### CODE REVIEW

File: src/test/java/UsersPutPatchTest.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+public class UsersPutPatchTest extends BaseTest {
+
+    @BeforeAll
+    public void prepareTestUsersData() {
+    }
+
+    @Order(1)
+    @DisplayName("Scenario #1. Task 50. Send PUT/PATCH with full body.")
+    @Test()
+    void usersPutPathGoldenPathTest() {
```
Comment (KirKarpov) [inline src/test/java/UsersPutPatchTest.java]: Looks like 'c' missed in the name PutPath. Same for other method names in this class

File: src/test/java/UsersPutPatchTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        // assert that the changed user is not removed from the application
+        assertTrue(clientUsers.sendGetUsers().contains(userToChangePut));
+        // assert that the updated user is not added to application
+        assertFalse(clientUsers.sendGetUsers().contains(userToUpdatePut));
+
+        // PATCH
+        // create a new user that will be changed through PATCH endpoint.
+        clientUsers.sendPostUsers(userToChangePatch);
+        // update the user through PATCH endpoint
+        assertEquals(400, clientUsers.sendPatchUsers(userToChangePatch, userToUpdatePatch));
```
Comment (KirKarpov) [inline src/test/java/UsersPutPatchTest.java]: You should check 409 status code according to the scenario

File: src/main/java/com/coherensolutions/rest/training/http/MyHttpDelete.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.coherensolutions.rest.training.http;
+
+import org.apache.http.client.methods.HttpPost;
+
+public class MyHttpDelete extends HttpPost {
```
Comment (KirKarpov) [inline src/main/java/com/coherensolutions/rest/training/http/MyHttpDelete.java:5]: Interesting solution :)
Usually people extends HttpEntityEnclosingRequestBase.

**Review by KirKarpov — APPROVED** [review]

File: src/test/java/UsersPutPatchTest.java (hunk 1 above)
Comment (artsiomMandrukevich) [reply src/test/java/UsersPutPatchTest.java]: done

File: src/test/java/UsersPutPatchTest.java (hunk 2 above)
Comment (artsiomMandrukevich) [reply src/test/java/UsersPutPatchTest.java]: done

**Pull request merged**